contrib/win32/win32compat/misc.c

```diff
@@ -1466,3 +1466,77 @@ is_absolute_path(char *path)
 	
 	return retVal;
 }
+
+/* return 0 - in case of failure */
+int
+create_directory_withsddl(char *path, char *sddl)
+{
+	struct stat st;
+	if (stat(path, &st) < 0) {
```

Comment:
how about if the directory exists but the sddl is incorrect?

Reply:
this can return -1 when out of memory (see implementation of fileio_stat).

Reply:
removed fatal and returning 0 in all cases so that caller can take action.